src/ngraph_api.cc

```diff
@@ -59,18 +69,25 @@ void Enable() { _is_enabled = true; }
 void Disable() { _is_enabled = false; }
 bool IsEnabled() { return _is_enabled; }
 
-size_t BackendsLen() { return ListBackends().size(); }
+size_t BackendsLen() { return BackendManager::GetNumOfSupportedBackends(); }
+
 vector<string> ListBackends() {
-  return ngraph::runtime::Backend::get_registered_devices();
+  auto supported_backends = BackendManager::GetSupportedBackendNames();
+  vector<string> backend_list(supported_backends.begin(),
+                              supported_backends.end());
+  return backend_list;
```

Comment:
Yes, nGraph core returns a vector, BackendManager maintains a set (so that we can do IsSupported in O(1)) and the python/C api returns a list/vector.